cocotb/share/lib/vpi/VpiImpl.cpp

```diff
@@ -296,13 +296,46 @@ GpiObjHdl* VpiImpl::native_check_create(void *raw_hdl, GpiObjHdl *parent)
 GpiObjHdl* VpiImpl::native_check_create(std::string &name, GpiObjHdl *parent)
 {
     vpiHandle new_hdl;
+    const vpiHandle parent_hdl = parent->get_handle<vpiHandle>();
     std::string fq_name = parent->get_fullname() + "." + name;
-    std::vector<char> writable(fq_name.begin(), fq_name.end());
-    writable.push_back('\0');
 
-    new_hdl = vpi_handle_by_name(&writable[0], NULL);
+    new_hdl = vpi_handle_by_name(const_cast<char*>(fq_name.c_str()), NULL);
```

Comment:
One thing I've always wondered is why are we looking up with the full name form the root scope and not passing the parent object as the second argument and just using the element name? Maybe it has something to do with mixed language support?

Reply:
Also, this should be safe since the `std::string` is non-const. Good catch.

Reply:
See #813. The full name might be worse for mixed language.

Also with C++11, we get contiguous chars, which may be the original reason for using a vector.

Reply:
Yeah, that is why I brought it up. I'm not certain how iterating with `vpi_iterate` over a handle retrieved with the FLI would work... that's my concern. I don't think the VPI, VHPI, or FLI were designed with mixed language cosimulation in mind, so I'm sure support is all over the place.

Reply:
Perhaps we can check how the handle was retrieved? If it is the same interface as we are currently using, we can use parent. Otherwise, we build the full name (and cache it) and look up with the other registered implementations from the root scope?

Reply:
We need more thorough tests for mixed language.
Looks like before e276c34c2e65ef9545430b14f1d86573a1853aea they used the parent handle for scope.

If the main issue is path format (`:` for VHPI, `/` for FLI, `.` for VPI), we could add a path conversion function between `GpiImplInterface`s for full name lookup?

Reply:
It might. I'm not too familiar with the issues in supporting mixed language sims, but it seems to work with Questa right now.